homeassistant/components/media_player/plex.py

```diff
@@ -272,93 +527,383 @@ def _active_media_plexapi_type(self):
-            return self._convert_na_to_none(self.session.title)
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            # find a string we can use as a title
+            title = None
+            if self.session:
+                if self._convert_na_to_none(self.session.title) is not None:
+                    title = self.session.title
+
+                    # append year for movies
+                    if self.media_content_type is MEDIA_TYPE_VIDEO:
+                        if self._convert_na_to_none(
+                                self.session.year) is not None:
+                            title = title + " (" + str(self.session.year) + ")"
+
+            return title
 
     @property
     def media_season(self):
         """Season of curent playing media (TV Show only)."""
-        return self._season
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.media_content_type is MEDIA_TYPE_TVSHOW:
+                if self.session is not None:
+                    if callable(self.session):
+                        return self._convert_na_to_none(
+                            self.session.seasons()[0].index).zfill(2)
+                    elif self._convert_na_to_none(
+                            self.session.parentIndex) is not None:
+                        return self.session.parentIndex.zfill(2)
+
+            return None
 
     @property
     def media_series_title(self):
         """The title of the series of current playing media (TV Show only)."""
-        from plexapi.video import Show
-        if isinstance(self.session, Show):
-            return self._convert_na_to_none(self.session.grandparentTitle)
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.media_content_type is MEDIA_TYPE_TVSHOW:
+                if self.session:
+                    return self._convert_na_to_none(
+                        self.session.grandparentTitle)
 
     @property
     def media_episode(self):
         """Episode of current playing media (TV Show only)."""
-        from plexapi.video import Show
-        if isinstance(self.session, Show):
-            return self._convert_na_to_none(self.session.index)
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.media_content_type is MEDIA_TYPE_TVSHOW:
+                if self.session:
+                    if self._convert_na_to_none(
+                            self.session.index) is not None:
+                        return str(self.session.index).zfill(2)
+
+    @property
+    def make(self):
+        """The make of the device (ex. SHIELD Android TV)."""
+        return self._make
 
     @property
     def supported_features(self):
         """Flag media player features that are supported."""
-        return SUPPORT_PLEX
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
```

Comment:
Every if check with no else clause or a limited one (like return None / 0) should become a guard clause.